go/vt/vtgate/scatter_conn.go

```diff
@@ -166,6 +167,10 @@ func (stc *ScatterConn) ExecuteMultiShard(
 		go stc.runLockQuery(ctx, session)
 	}
 
+	if session.Options != nil {
+		session.Options.FetchLastInsertId = fetchLastInsertID
+	}
+
 	allErrors := stc.multiGoTransaction(
```

Comment:
why did we break this up in `ExecuteMultiShard` and not do similar to `StreamExecuteMulti`?